tests/calculators/test_workflow.py

```diff
@@ -183,6 +184,83 @@ def test_not_nan(self, CalculatorClass, params, device, dtype):
                 torch.autograd.grad(energy, system[2], retain_graph=True)[0]
             ).any()
 
+    def test_smearing_incompatability(self, CalculatorClass, params, device, dtype):
+        """Test that the calculator raises an error if the potential and calculator are incompatible."""
+        if type(CalculatorClass) in [EwaldCalculator, PMECalculator, P3MCalculator]:
+            params["smearing"] = None
+            with pytest.raises(
+                TypeError, match="Must specify smearing to use a potential with .*"
+            ):
+                CalculatorClass(**params)
+
+    def test_periodicity_incompatability(
+        self,
+        CalculatorClass,
+        params,
+        device,
+        dtype,
+    ):
+        if CalculatorClass in [PMECalculator, P3MCalculator, EwaldCalculator]:
+            charges, cell, positions, neighbor_indices, neighbor_distances = (
+                self.cscl_system(device=device, dtype=dtype)
+            )
+            match = (
+                "K-space summation is not implemented for 1D or non-periodic systems."
+            )
+            with pytest.raises(ValueError, match=match):
+                CalculatorClass(**params).to(device).forward(
+                    charges=charges,
+                    cell=cell * 4,
+                    positions=positions,
+                    neighbor_indices=neighbor_indices,
+                    neighbor_distances=neighbor_distances,
+                    periodic=torch.tensor([True, False, False], device=device),
+                )
+            match = r"Maximum distance along non-periodic axis \(.*\) exceeds one third of cell size \(.*\)\."
+            with pytest.raises(ValueError, match=match):
+                CalculatorClass(**params).to(device).forward(
+                    charges=charges,
+                    cell=cell,
+                    positions=positions,
+                    neighbor_indices=neighbor_indices,
+                    neighbor_distances=neighbor_distances,
+                    periodic=torch.tensor([True, True, False], device=device),
+                )
+
+    def test_periodicity_true_value(
+        self,
+        CalculatorClass,
+        params,
+        device,
+        dtype,
+    ):
+        """Test that values coincide with values from LAMMPS"""
+        true_value = -383.44635
```

Comment:
I don't want to overdo these tests but maybe we can do a better test as just a regression one.

I will think about it.